xarray/core/formatting.py

```diff
@@ -804,10 +806,18 @@ def extra_items_repr(extra_keys, mapping, ab_side, kwargs):
 
             if compat == "identical" and is_variable:
                 attrs_summary = []
+                a_attrs = a_mapping[k].attrs
+                b_attrs = b_mapping[k].attrs
 
+                attrs_to_print = set(a_attrs) ^ set(b_attrs)
+                for attr in set(a_attrs) & set(b_attrs):
+                    if a_attrs[attr] != b_attrs[attr]:
+                        attrs_to_print.update([attr])
```

Comment:
I wonder if this would be better:
```suggestion
                attrs_to_print.update(
                    {k for k in set(a_attrs) & set(b_attrs) if a_attrs[k] != b_attrs[k]}
                )
```
which would require fewer update calls?